### ISSUE TITLE
Example CRA repo fails

### ISSUE THREAD

**Issue by jeffmcaffer** [microsoft/fluentui#15864]
### Environment Information

Just following current website instructions on an up to date Windows 10 machine

### Please provide a reproduction of the bug in a codepen:

1. Create a new CRA based setup by following the Getting started info 
https://developer.microsoft.com/en-us/fluentui#/get-started#option-2-create-react-app
2. Run `npm start`
3. Observe that the following error in the spawned browser
```
 C:/git/my-app/node_modules/jest-diff/build/diffLines.d.ts
TypeScript error in C:/git/my-app/node_modules/jest-diff/build/diffLines.d.ts(8,13):
'=' expected.  TS1005

     6 |  */
     7 | import { Diff } from './cleanupSemantic';
  >  8 | import type { DiffOptions } from './types';
       |             ^
     9 | export declare const diffLinesUnified: (aLines: Array<string>, bLines: Array<string>, options?: DiffOptions | undefined) => string;
    10 | export declare const diffLinesUnified2: (aLinesDisplay: Array<string>, bLinesDisplay: Array<string>, aLinesCompare: Array<string>, bLinesCompare: Array<string>, options?: DiffOptions | undefined) => string;
    11 | export declare const diffLinesRaw: (aLines: Array<string>, bLines: Array<string>) => Array<Diff>;
```

#### Actual behavior:

See above

#### Expected behavior:

Site should come up in browser

### Priorities and help requested:

I'm just getting started with Fluent UI and this blocks that effort. Note that the other path of adding Fluent to an existing CRA appears to also be broken as CRA has moved to React 17 and Fluent is still back on 16. So as it is there is no way for a new Fluent user to get started.

**Comment by jeffmcaffer** [comment]
A little investigation found that perhaps this is an issue with TS support for type importing (see https://stackoverflow.com/questions/61783437/react-typescript-error-on-build-in-node-modules-jest-diff-build-difflines-d-ts) which recommends updating TS to >= 3.8. I updated to 3.9.7 (saw that in some other post) and the example site comes up but I get the following warning

```
You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0

YOUR TYPESCRIPT VERSION: 3.9.7

Please only submit bug reports when using the officially supported version.
```

And while I'm not sure what the app should look like, it appears to have a broken logo image as shown below. Could be unrelated...

![image](https://user-images.githubusercontent.com/10070956/98489052-337b2a00-21e1-11eb-824a-da1211f3fbad.png)

**Comment by jeffmcaffer** [comment]
Digging even further it seems that that example repo is very old and still references `"office-ui-fabric-react": "^6.174.0"`. 

I suggest you either update the example repo or delete the reference from the doc so as to not mislead folks to using old libraries. Frankly the idea of using a pre-created CRA app and Git repo was not all that exciting. You'd get a random CRA setup created some time ago and perhaps (or not) updated.

IMHO it would be much better if the baseline was just "do what you do to setup CRA and then do XXX to get Fluent UI installed". Unfortunately, that path is also currently broken because, as noted ☝️ , the latest CRA uses React 17.

Perhaps the doc should be updated to outline the latest version of CRA is supported.

**Comment by paulgildea** [comment]
@jeffmcaffer Thanks for digging into this issue!

It's currently on our list to address our getting started experience - #15776 @ecraig12345 / @joschect FYI.

Since you've done a lot of investigation in this area, would you be wiling to contribute a fix the current getting started? If not now worries just thought I would ask 😄 .

@JustSlone FYI.

**Comment by paulgildea** [comment]
@ecraig12345 / @joschect Feel free to resolve as dupe if you've got this part covered in the current work item for updating getting started templates.

**Comment by jeffmcaffer** [comment]
I could contribute a PR but I wonder about the path here. As I mentioned above, Having a repo that folks clone that already has CRA setup feels less than optimal. As CRA changes it's pretty much black box as to what's been done in there before the user clones the repo. For just messing around and trying out Fluent sure but not as the basis for my new app. Better IMHO would be to make it easy to just do `create-react-app` and `npm install @fluentui/react`. Paste in a simple App body from the doc and shazam ⚡ .

I eventually worked out that latter path but it's not pretty as you have to CRA, downgrade React, install TS and all the @types, install @fluentui, and a few more shenanigans I don't remember. Even then I still have not found things like IComponentStyle.

the cited issue may address the main topic here (getting started is very hard) so this can be closed as a dupe if you like. I'll comment over there on some thoughts having just tried to get started.

**Comment by ecraig12345** [comment]
I noticed this recently as well, that our templates are super outdated. @joschect was actually working last week during our "fix hack learn" week on figuring out a better solution but not sure if he landed on anything yet. If possible we wanted something where we could host a template within our repo (so it's more likely to stay up to date!), but we'd prefer not to pull the whole weight of `react-scripts` into our local dependency graph. A copy paste solution like you mentioned @jeffmcaffer could be one workaround.

Regarding the React version issue, we're currently investigating React 17 support for version 8 (see #15732) and would rather not officially add it to our peer dep range quite yet. In the meantime as a workaround you can either:
- Use `react-scripts` version 3 (not 4), or
- Ignore the invalid peer dependency warning

(One other background point, if I remember right the reason why we originally had a CRA setup where you had to clone a repo was because CRA didn't support templates at the time. And I guess whoever set up that repo preferred not to do the copy-paste setup.)

**Comment by jeffmcaffer** [comment]
Fair enough (and thanks for the additional context). The repo clone approach is certainly not the end of the world and does have some merits. I think I'd probably use that to get going quickly (fewer steps in theory) and then once I decided that Fluent UI was for me, I'd start from scratch and copy over what I wanted (if anything). If you are going for that, perhaps you can use [GitHub repo templates](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/creating-a-template-repository).

I encourage you to keep the "add fluent ui to an existing app" path active and first class. For that, restrictions on <= react 16.* and react-scripts < 4 and TS < 3.8, ... all need to be surfaced up front. Just installing CRA and TS and React anew will result in literally all of them being incompatible with Fluent as of today. Tips for users as to how they can either downgrade or install the older versions would be great.

**Comment by ecraig12345** [comment]
Thanks for the suggestions, that all makes sense. Good idea about the github repo templates (I think those are relatively new too).

You actually should be fine using TS 4 with `@fluentui/react`--our restriction is that you can't go *lower* than 3.7. If something isn't working with a newer TS version you can file an issue and we'll look into it.

**Comment by jeffmcaffer** [comment]
LOL yeah that TS versioning thing is emblematic of the situation. Template repo has `"typescript": "3.4.5"`, tools say `SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0`, you say `>= 3.7`, the app only works with `>= 3.8` (as best I can tell and despite the tool complaints), and a fresh CRA install gets you `4.0.5` (IIRC).

Ah, life in the middle of the software stack...

**Comment by Hotell** [comment]
Hi folks, I played with it a bit

> it appears to have a broken logo image 

path to that image is non existent within the repo that is used. Dunno what was the original intent ( to render fabric ui logo instead of default React ? )

> the following warning
```
You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0

YOUR TYPESCRIPT VERSION: 3.9.7

Please only submit bug reports when using the officially supported version.
```

- that is "a greeting" for eslint (to be precise typescript-eslint parser). Everything should work without issues though.

>  still references "office-ui-fabric-react": "^6.174.0".

- yeah that needs to migrate to `@fluent-ui/react`

---

### Quick Workaround

-  bump react-scripts to `3.4.4` and typescript to `3.8.3` and all dev related errors will be gone :)

### Quick fixes

1. I can bump those dev dependencies via PR to unblock these kind of issues
2. then attempt to migrate whole project to `@fluent-ui/react` 

WDYT @JustSlone @ecraig12345 ?

**Comment by JustSlone** [comment]
@Hotell, what do you think about converting this to a [CRA template](https://create-react-app.dev/docs/custom-templates/)? 

Your choice if you want to make the quick fixes to get this off the ground first, or just jump right in to converting to a template. 

I would recommend doing the conversion in the same repo and renaming the repo to not break links, but if that is a pain, a new repo is fine.

**Comment by Hotell** [comment]
I'll ping authors of recent PRs that make sense to merge and get those in. Besides those I'll fix other issues by my own so it works for now without any need to change the website docs etc.

 > what do you think about converting this to a CRA template?

That's the best approach going forward IMO  (there are some existing ones already!  https://github.com/nissan/cra-template-ts-fabric#readme ). I'd suggest to create that template under this repo so we have things collocated, WDYT ? When that is done I'd update the web getting started docs and archive that existing CRA repo.

**Comment by Hotell** [comment]
All mentioned issues should be resolved now. Please update your CRA repo with latest changes that we shipped into master.  I'm gonna close this. Feel free to re-open if you'll still experience reported issues. thank you
